# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/mediasource/ResourceQueue.h
# Commit: f6df4eb84932
# Full Hash: f6df4eb84932ddaccae4e53d4897234e0153913e
# Author: Chris Pearce <cpearce@mozilla.com>
# Date: 2019-06-14 10:00:22
# Regressor Bug: 1558364
# File Overlap Count: 1
# Description:
#   Bug 1558364 - Add MediaSpan and use it for TrackBuffersManager::mInputBuffer. r=jya
#   
#   As seen in this profile of a Twitch replay: https://perfht.ml/2K9Ydb3 we can
#   often end up spending time in TrackBuffersManager::CodedFrameProcessing()
#   shaving off bytes from the front off TrackBuffersManager::mInputBuffer. This
# ==============================================================================

diff -r e708fe457a95 -r f6df4eb84932 dom/media/mediasource/ResourceQueue.h
--- a/dom/media/mediasource/ResourceQueue.h	Fri Jun 14 00:37:25 2019 +0000
+++ b/dom/media/mediasource/ResourceQueue.h	Fri Jun 14 02:10:09 2019 +0000
@@ -8,7 +8,7 @@
 #define MOZILLA_RESOURCEQUEUE_H_
 
 #include "nsDeque.h"
-#include "MediaData.h"
+#include "MediaSpan.h"
 
 namespace mozilla {
 
@@ -26,9 +26,9 @@
 // timepoint.
 
 struct ResourceItem {
-  ResourceItem(MediaByteBuffer* aData, uint64_t aOffset);
+  ResourceItem(const MediaSpan& aData, uint64_t aOffset);
   size_t SizeOfIncludingThis(MallocSizeOf aMallocSizeOf) const;
-  RefPtr<MediaByteBuffer> mData;
+  MediaSpan mData;
   uint64_t mOffset;
 };
 
@@ -46,13 +46,13 @@
   // Copies aCount bytes from aOffset in the queue into aDest.
   void CopyData(uint64_t aOffset, uint32_t aCount, char* aDest);
 
-  void AppendItem(MediaByteBuffer* aData);
+  void AppendItem(const MediaSpan& aData);
 
   // Tries to evict at least aSizeToEvict from the queue up until
   // aOffset. Returns amount evicted.
-  uint32_t Evict(uint64_t aOffset, uint32_t aSizeToEvict, ErrorResult& aRv);
+  uint32_t Evict(uint64_t aOffset, uint32_t aSizeToEvict);
 
-  uint32_t EvictBefore(uint64_t aOffset, ErrorResult& aRv);
+  uint32_t EvictBefore(uint64_t aOffset);
 
   uint32_t EvictAll();
 